src/GameLibs/Snake/System/MoveDirection.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+** EPITECH PROJECT, 2023
+** Arcade
+** File description:
+** MoveDirection
+*/
+
+#include <iterator>
+#include <memory>
+#include <vector>
+#include "Forward.hpp"
+#include "MoveDirection.hpp"
+#include "MagicValue.hpp"
+#include "SnakeCompType.hpp"
+#include "ChangeDir.hpp"
+#include "Sprite.hpp"
+#include "MoveInput.hpp"
+
+static void checkHitChangeDir(std::shared_ptr<Snake::Component::ChangeDir> changeDir, std::shared_ptr<Arcade::ECS::IEntity> entity)
+{
+    auto bodySpriteComps = entity->getComponents(Arcade::ECS::CompType::SPRITE);
+
+    for (auto &bodySpriteComp : bodySpriteComps) {
+        auto bodySprite = std::static_pointer_cast<Arcade::Graph::Sprite>(bodySpriteComp);
+        if (bodySprite->pos.x > changeDir->pos.x + (CASE_SIZE_WIDTH / 3) ||
+                bodySprite->pos.x < changeDir->pos.x - (CASE_SIZE_WIDTH / 3)) {
+            continue;
+        }
+        if (bodySprite->pos.y > changeDir->pos.y + (CASE_SIZE_HEIGHT / 3) ||
+                bodySprite->pos.y < changeDir->pos.y - (CASE_SIZE_HEIGHT / 3)) {
+            continue;
+        }
+        auto &bodyDir = static_cast<Snake::Component::Forward &>(entity->getComponents(MOVEABLE_KEY));
+        auto nextCase = Snake::System::toNextCase(bodySprite->pos, bodyDir.direction);
+        bodySprite->pos.x = nextCase.x;
+        bodySprite->pos.y = nextCase.y;
+        bodyDir.direction = changeDir->direction;
+    }
+}
+
+void Snake::System::MoveDirection::run(float deltaTime, Arcade::ECS::IEventManager &eventManager, Arcade::ECS::IEntityManager &entityManager)
+{
+    auto snake = entityManager.getEntitiesById(SNAKE_HEAD);
+    auto directionsEntities = entityManager.getEntitiesByComponentType(Arcade::ECS::CompType::CHANGEDIR);
```

Comment:
y'a que le snake qui a des CHANGEDIR pas besoin de prendre une liste

Reply:
ducoup mtn je get plus le snake, que les forward, donc ça revient au même

Reply:
mais le premier for il sert toujours à rien dcp t'ittere sur une liste qu'a forcément qu'un élement